test/kt/two_pass_scan.cpp

```diff
@@ -67,30 +67,58 @@ generate_scan_data(T* input, std::size_t size, std::uint32_t seed)
     }
 }
 
-template <typename T, sycl::usm::alloc _alloc_type, typename BinOp, typename KernelParam>
```

Comment:
Can we just pass one Kernel Param and then use another layer of `get_new_kernel_params` within this function?
This seems more modular and extendable.

Reply:
Yes, I have made this change.